.github/docker/rockylinux-8.Dockerfile

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2023 Intel Corporation
+# Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+# See LICENSE.TXT
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#
+# Dockerfile - a 'recipe' for Docker to build an image of rockylinux-based
+#              environment for building the Unified Runtime project.
+#
+
+# Pull base image
+FROM registry.hub.docker.com/library/rockylinux:8
+
+# Set environment variables
+ENV OS rockylinux
+ENV OS_VER 8
+ENV NOTTY 1
+
+# Additional parameters to build docker without building components.
+# These ARGs can be set in docker building phase and are used
+# within bash scripts (executed within docker).
+ARG SKIP_DPCPP_BUILD
+ARG SKIP_LIBBACKTRACE_BUILD
+
+# Base development packages
+ARG BASE_DEPS="\
+	cmake \
+	git \
+	glibc-devel \
+	libstdc++-devel \
+	make"
+
+# Unified Runtime's dependencies
+ARG UR_DEPS="\
+	doxygen \
+	python3 \
+	python3-pip"
+
+# Packages required by requirements.txt
+ARG PRE_PYTHON_DEPS="\
+	libjpeg-turbo-devel \
+	python3-devel \
+	python3-wheel \
+	zlib-devel"
+
+# Miscellaneous for our builds/CI (optional)
+ARG MISC_DEPS="\
+	clang \
+	ncurses-libs-6.1 \
+	passwd \
+	sudo \
+	wget"
+
+# Update and install required packages
+RUN dnf update -y \
+ && dnf --enablerepo devel install -y \
+	${BASE_DEPS} \
+	${UR_DEPS} \
+	${PRE_PYTHON_DEPS} \
+	${MISC_DEPS} \
+ && dnf clean all
+
+# Additional dependencies (installed via pip)
+COPY third_party/requirements.txt requirements.txt
+# Comment out clang-format as it's not working properly on this distro
```

Comment:
Do we have to file a new issue for fixing this problem?

Reply:
I guess so. If we decide to merge this PR we should probably track this - either we should fix the weird problems in clang-format installation (via pip) on rockylinux8 or update the version of clang-format...

Reply:
We decided we won't test on regular basis python dependencies in some OSes, due to missing packages in some of these repos (in their pip database to be precise).

Development checks will be run only on selected OS, in our case mostly on Ubuntu 22.04.